btk/metrics.py

```diff
@@ -1,484 +1,497 @@
+
+        noise_threshold = get_mean_sky_level(survey, survey.filters[self.meas_band_num])
+        if "catalog" not in measure_results.keys():
+            meas_func = measure_results.keys()
+            metrics_results = {}
+            for f in meas_func:
+                metrics_results_f = compute_metrics(
+                    blend_results["blend_images"],
+                    blend_results["isolated_images"],
+                    blend_results["blend_list"],
+                    measure_results[f]["catalog"],
+                    measure_results[f]["segmentation"],
+                    measure_results[f]["deblended_images"],
+                    self.use_metrics,
+                    noise_threshold,
+                    self.meas_band_num,
+                    target_meas,
+                    blend_id_start=self.blend_counter,
+                    dim_order=self.measure_generator.dim_order,
+                )
+                metrics_results[f] = metrics_results_f
+
+        else:
+            metrics_results = compute_metrics(
+                blend_results["blend_images"],
+                blend_results["isolated_images"],
+                blend_results["blend_list"],
+                measure_results["catalog"],
+                measure_results["segmentation"],
+                measure_results["deblended_images"],
+                self.use_metrics,
+                noise_threshold,
+                self.meas_band_num,
+                target_meas,
+                blend_id_start=self.blend_counter,
+                dim_order=self.measure_generator.dim_order,
             )
-        true_table, detected_table, detection_summary = evaluate_detection(
-            batch_detection_result[0], batch_detection_result[1], batch_index=i
-        )
-        results["detection"][0] = astropy.table.vstack([results["detection"][0], true_table])
-        results["detection"][1] = astropy.table.vstack([results["detection"][1], detected_table])
-        results["detection"][2].extend(detection_summary)
-        # Evaluate segmentation algorithm
-        segmentation = metrics_params.get_segmentation()
-        results["segmentation"].append(evaluate_segmentation(segmentation, index=i))
-        # Evaluate flux measurement algorithm
-        flux = metrics_params.get_flux()
-        results["flux"].append(evaluate_flux(flux, index=i))
-        # Evaluate shape measurement algorithm
-        shapes = metrics_params.get_shapes()
-        results["shapes"].append(evaluate_shapes(shapes, index=i))
-    return results
+
+        self.blend_counter += len(blend_results["blend_list"])
+        return blend_results, measure_results, metrics_results
+
+
+def run_metrics(metrics_generator: MetricsGenerator, n_batches=100):
+    """Uses a `metrics_generator` objec to summarize metrics results for `n_batches` batches."""
+    measure_funcs = metrics_generator.measure_generator.measure_functions
```

Comment:
that line feels weird..

Reply:
Agreed. I don't know about this whole run_metrics function...